### ISSUE TITLE
[BUG]:ISSUES related to the use of SPLIT

### ISSUE THREAD

**Issue by Gary-K9RX** [ramdor/Thetis#59]
-Puresignal (sometimes) locks off after having used SPLIT (already discussed previously where you were able to see it happen)
-MON is left on sometimes after using SPLIT/CW and then, having turned SPLIT off, going to SSB on that same band
-Possibly eliminate issues related to the use of SPLIT by being sure IF it is inadvertently left on when changing a band that it is the first thing that happens in that band change - check to see if SPLIT is on and if it is turn it off? Perhaps that will circumvent the above issues.

**Comment by ramdor** [comment]
cant replicate 1 or 2 in this current version unfortunately 2.9.0.8. If still an issue we can re-open, and perhaps you can give a video of it happening Gary. As it happens there is an option to turn of split on band change already :

![image](https://github.com/ramdor/Thetis/assets/3124967/ca3b2992-8695-475b-87f4-21005abd768b)

**Comment by Gary-K9RX** [comment]
Richie,

Unfortunately I happen to be out of town when you sent this - I'm here, 
all the time (work from home) ... but was gone to pick up grandkids.

I can't find the issue - it appears you closed it. It was not a function 
of having the "disable split on band change"! I have that. I don't 
remember how I worded it but it was then and is now still a current 
problem. I'm happy to assist in resolving it. I can make it happen quite 
frequently.


Gary
K9RX




------ Original Message ------
From "Richie" ***@***.***>
To "ramdor/Thetis" ***@***.***>
Cc "Gary-K9RX" ***@***.***>; "Author" ***@***.***>
Date 6/19/2023 1:49:35 PM
Subject Re: [ramdor/Thetis] [BUG]:ISSUES related to the use of SPLIT 
(Issue #59)

**Comment by ramdor** [comment]
Hi gary, the report is at the top.

-Possibly eliminate issues related to the use of SPLIT by being sure IF it is inadvertently left on when changing a band that it is the first thing that happens in that band change - check to see if SPLIT is on and if it is turn it off? Perhaps that will circumvent the above issues.

Is what you said, which is exactly the same as using that option :)

R.

**Comment by Gary-K9RX** [comment]
Richie,
The first issue related to split and the use of PS you had confirmed a few months ago. It took you a while to get it to do it but you managed to see it once you found it. I can look for that discussion on the Thetis forum and try to get you back up to speed. It is real for sure (as noted: you saw it as well). 

The 2nd issue - I just tried it here and this is what I do to see it: Go to (say) 20M CW. Now hit MULTI RX, A>B and SPLIT in that order. Tune VFO B up - transmit - all working fine (split), MON is of course on. Now hit SPLIT and MULTI to turn them off. The VFO goes back to showing just A. Now hit 20 again to get to an SSB stack. MON is still on. (this is using a midi panel, and the check box you mentioned is indeed checked, I confirmed that again). 

I just did the scenario you asked me to check and indeed when I again turn it on, as described above, then don't turn it off but just do a band change (to a band where the currently selected band stack is SSB) - SPLIT goes off, MULTI remains on (!) - MON remains on as well. 

Gary

**Comment by ramdor** [comment]
1) ok, will try to track it down when I have a mo

2) mon stays on if switched on in ssb mode afaik, but if your cw mode uses sidetone, then mon will only switch on in cw. Go to non cw mode and turn mon off. See if this helps

3) multi rx will auto switch off if it is outside the sample rate centred around the vfo. So if you step through band stacks and the multi rx is too far away it will switch off

I would make a vid showing points 2+3, as I don't think they are bugs tbh, but short of time atm.

**Comment by ramdor** [comment]
Tracked the tread down, added here for reference : https://community.apache-labs.com/viewtopic.php?f=20&t=3739

Also, for reference issue #133

**Comment by Gary-K9RX** [comment]
Richie,

Apologies again - my grand kids are visiting which is why I missed the 
first question you gave and then you closed #59 (and thanks again for 
reopening it btw)... and because they are here and haven't ever seen the 
ocean we are going on a short holiday starting in the morning, we'll be 
back on Friday. I very much want these things resolved ... but I'll be 
out of touch for that time period. This is the first time (times, 2 now) 
that I've been away from the house for a good year... but it is what it 
is.

Gary


------ Original Message ------
From "Richie" ***@***.***>
To "ramdor/Thetis" ***@***.***>
Cc "Gary-K9RX" ***@***.***>; "Author" ***@***.***>
Date 6/20/2023 1:02:01 PM
Subject Re: [ramdor/Thetis] [BUG]:ISSUES related to the use of SPLIT 
(Issue #59)

**Comment by Gary-K9RX** [comment]
Richie,

Just info in response... I understand you'll look at this when you get a 
chance.

I don't use MON on when in any mode other than CW. Thus my noticing the 
fact that it remains on under this circumstance. Of necessity I have 
been forced to 'turn it off' when I found I did this move - worked a 
station, split, on CW and then happen to go to another band stack - 
either SSB or DIG only to be surprised when I hear my voice or the 
annoying FT8 tones. (it does however work - sometimes... so there's 
that)

Just as an aside:
It would be nice if SPLIT could be programmed to (either automatically 
or via a selection check box in SETUP):
a) turn on MULTI
b) set VFO B at a particular offset (I had another suggestion about this 
- having several choices, displayed in the VFO B box, for split 
separation depending on mode, 1, 2, 3, 5 for CW and 5 / 10 for SSB)
c) and turn on SPLIT
.... and then in response undo all of these.

MULTI might be designed to do as you say below (#3 point), I don't doubt 
that of course - but functionally at least relative to its use in split 
the above assemblage and dis-assemblage would be more "proper". It is 
basically what a Yaesu or Icom rig would do if one hit SPLIT, instead of 
here, having to do 3 different steps.

g.




------ Original Message ------
From "Richie" ***@***.***>
To "ramdor/Thetis" ***@***.***>
Cc "Gary-K9RX" ***@***.***>; "Author" ***@***.***>
Date 6/19/2023 6:49:55 PM
Subject Re: [ramdor/Thetis] [BUG]:ISSUES related to the use of SPLIT 
(Issue #59)

**Comment by ramdor** [comment]
closing this as not sure what applies anymore with newer version. Quick split has been added etc. Please could you add another issue if you find any Gary, cheers.

**Comment by Gary-K9RX** [comment]
I didn't see anything in the list of changes for the latest release(s) 
that relates to issue 59 "the use of split". There's full reason to 
think it all still exists.

Its annoying as all get out - quite frequently when split is used the 
MON button stays on (if it were SPLIT and CW mode, which it often is). 
This means if I next use it on SSB or DIG that I hear myself and have to 
quickly shut it off or on DIG/FT8 the tones are there. Less frequent but 
still often enough to be a pain, when using SPLIT and then turning it 
off and going to a SSB - Puresignal stops working and I have to close 
the program and restart it (you've seen this yourself in the past).

Gary
K9RX




------ Original Message ------
From "Richie" ***@***.***>
To "ramdor/Thetis" ***@***.***>
Cc "Gary-K9RX" ***@***.***>; "Author" ***@***.***>
Date 10/25/2023 9:09:50 AM
Subject Re: [ramdor/Thetis] [BUG]:ISSUES related to the use of SPLIT 
(Issue #59)

**Comment by ramdor** [comment]
#59 is a list of different issues, which is not the idea of the issue tracker. Each issue should either be entered as a bug, or as an enhancement.  The split improvement that auto offsets and included multirx has been addressed and was listed in #59.  The issue regarding PS failing after CW is uses has its own issue #226 and is being actively worked on by Warren.

In CW MON will be on IF you have the checkbox Sidetone enabled. If you turn mon OFF when in CW then the sidetone checkbox will also disable. The state of MON is entirely mirrored by the Sidetone checkbox. When in any mode other than CW, the the state of MON is independent of the the CW sidetone state. So if you turn mon off in SSB, then move to CW where sidetone is on, the MON will auto enable. When you return to a mode other than CW the mon state will change if needed.  This is independent of split.  If this is not working like this for you please send me a copy of your database.xml and a quick video of it happening.

See vid: https://youtu.be/Y7yXknto7_I

There are issues regarding MON when running RX2 and a split, especially when RX2 is running as a different mode, but those are far reaching and not just related to CW/sidetone.

**Comment by ramdor** [comment]
and just to add. I do not use CW so have to go on reports regarding CW issues from others. There may be edge cases where this MON button is annoying and shows when it should not, but as I don't use CW it is not something I would get to see. The lack of others reporting this with MON makes me think it is perhaps an issue local to your install.

**Comment by ramdor** [comment]
so after all my diatribe and testing, I decided to do a code review around the MON state recovery and I did find an issue. CWU may have incorrectly recovered other mode MON states. This will be resolved in [2.10.3]

**Comment by Gary-K9RX** [comment]
Richie,

Thank you. I'll look at parsing them out. That stated both are related 
to and happen when using split - thus the combination of the 2 listed. 
Both are listed as a BUG.

[you just posted that you now see this and confirm this is "real" - but 
as a point of explanation I'll leave this - read or don't as you like]
As for CW MON. I realize how it is suppose to work. I don't have MON on 
for any mode other than CW. However as noted when you use Split 
operation in CW mode it is not unusual, after I'm done, I've turned 
MULTI RX and SPLIT off and I've gone to say SSB, its not unusual for the 
MON now to be on! Thus a bug as this is not how I had it set. It does 
the same thing, at times, when repeating this but going to DIG/FT8. 
Similarly PureSignal being turned off happens under the same 
circumstances and it too doesn't always happen. As a matter of fact we 
had talked about this and at one time you were unable to get it to do 
this - and then finally you reported you were able to see it (apparently 
that now is issue #226) thus acknowledging it was real and happened as I 
had noted.

Reading #226 I'm a bit confused here. I reported this on the forum. Then 
you added the issues list so I reported it there, as issue #59. Again - 
since both effects I'm seeing, CW MON/PS, are related to the same action 
usage, SPLIT, I thought it appropriate to keep/list them together so 
they could be worked on at the same time. Now I see part of that - the 
Split/Puresignal issue, being listed as a separate item - no mention of 
it previously being mentioned (#59), the discussion there, or my 
involvement (which I mention only because I'm here to help if needed). 
As a note related to #226: I did and do see it on an 8000. Three times 
over the last 2 days alone!

I have a few FEATURE listings :  #74, 75, 82. Let me know please what 
help I can be or questions I can answer on them.

I'm glad to hear Warren is working on the PS/Split issue. Thank you for 
that info.


Gary




------ Original Message ------
From "Richie" ***@***.***>
To "ramdor/Thetis" ***@***.***>
Cc "Gary-K9RX" ***@***.***>; "Author" ***@***.***>
Date 10/25/2023 10:31:20 AM
Subject Re: [ramdor/Thetis] [BUG]:ISSUES related to the use of SPLIT 
(Issue #59)
